plugin/restore/mysql/mysql.go

```diff
@@ -310,7 +310,13 @@ func (r *Restore) GetLatestBackupBeforeOrEqualTs(ctx context.Context, backupList
 	}
 
 	var eventTsList []int64
+	var backupWithValidBinlogInfoList []*api.Backup
 	for _, b := range backupList {
+		if b.Payload.BinlogInfo.IsEmpty() {
+			log.Debug("Skip parse binlog event timestamp of backup which binlogInfo field in payload field is empty", zap.Int("Id", b.ID))
+			continue
+		}
+		backupWithValidBinlogInfoList = append(backupWithValidBinlogInfoList, b)
 		eventTs, err := r.parseBinlogEventTimestamp(ctx, b.Payload.BinlogInfo)
```

Comment:
It's better to also defend in `parseBinlogEventTimestamp`: return error if binlogInfo is empty.
This could prevent future mistakes, because `--start-position 0` is a valid option for `mysqlbinlog`, and may produce transient but disastrous bugs (e.g., data corruption).

Reply:
In this case, we should only look at the backups that are completed successfully.